### PR TITLE
MaxPool2d: use channels_last format for both output and indice when input is channels_last

### PR DESCRIPTION
**Pull request by mingfeima**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61245 MaxPool2d: use channels_last format for both output and indice when input is channels_last**

Differential Revision: [D29557884](https://our.internmc.facebook.com/intern/diff/D29557884)

### CODE REVIEW

File: aten/src/ATen/native/DilatedMaxPool2d.cpp [hunk 1]
```diff
@@ -111,9 +111,6 @@ const Tensor& indices) {
   const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
   const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
```
Comment (mingfeima) [inline aten/src/ATen/native/DilatedMaxPool2d.cpp:113]: this check is a duplicate.

File: aten/src/ATen/native/DilatedMaxPool2d.cpp [hunk 2]
```diff
@@ -111,9 +111,6 @@ const Tensor& indices) {
...
 
-  TORCH_CHECK((input.ndimension() == 3 || input.ndimension() == 4),
-    "non-empty 3D or 4D (batch mode) tensor expected for input");
-
```
Comment (mingfeima) [inline aten/src/ATen/native/DilatedMaxPool2d.cpp:116]: this check is a duplicate.

**Comment by mingfeima** [comment]
Make sure that both `output` and `indice` follow the same memory format as `input`.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557884).

**Pull request closed without merging**